python/services/dataservice/dmod/dataservice/initial_data_adder_impl.py

```diff
@@ -64,7 +69,8 @@ def _add(path: Path, dest: Optional[str] = None, add_root: Optional[Path] = None
             elif not dest:
                 dest = path.relative_to(add_root)
 
-            if not self._dataset_manager.add_data(dataset_name=self._dataset_name, dest=dest, data=path.read_bytes()):
+            if not self._dataset_manager.add_data(dataset_name=self._dataset_name, dest=dest, data=path.read_bytes(),
```

Comment:
This seems like a great place to use the reader interface.
```suggestion
            with path.open("rb") as fp:
                if not self._dataset_manager.add_data(dataset_name=self._dataset_name, dest=dest, data=fp,
```

Reply:
Perhaps, but this is probably too far out of scope here.  Admittedly, I did modify calls to `self._dataset_manager.add_data` in the _InitialDataAdder_ implementations.  One could argue (questionably) that this bled into scope due to it being done as part of the other modifications to _BmiAutoGenerationAdder_, but, more importantly, those changes were to fix something that was broken.

Reply:
I think you could argue this is out of scope, but IMO the change is minor and the potential performance consequences are high. My concern is if a large file is passed as `path`. This will blow up the resident memory usage of the process b.c. the file read is greedy (likely a demand paged `mmap` that is `munmap`ed, but still not great). If we just pass something that has a `read()` method, `add_data` can perform buffered reads that have a lesser potential to degrade performance.